plugins/org.jkiss.dbeaver.model.lsp/src/org/jkiss/dbeaver/model/lsp/utils/URIUtils.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.lsp.utils;
+
+import org.jkiss.code.NotNull;
+
+import java.util.regex.Pattern;
+
+public class URIUtils {
+
+    private static final Pattern URI_PATTERN = Pattern.compile("^lsp://([A-Za-z0-9_-]+)/(\\S.*)$");
+
+    public static void validateUri(@NotNull String uri) {
+        if (!URI_PATTERN.matcher(uri).matches()) {
+            throw new IllegalArgumentException(
+                String.format("Invalid URI format: %s, expected lsp://{projectId}/{resourcePath}", uri)
+            );
+        }
+    }
+
+    @NotNull
```

Comment:
Let's create class DocumentURI and keep all parts there. Use regex instead of substrings to extract parts.